### PR TITLE
Improve the github CI by uploading logs of failed runs

### PR DESCRIPTION
**Pull request by holke**
The CI script now uploads configure and make logs as artifacts.

See my test runs and their artifacst https://github.com/holke/t8code/actions/runs/2301835043 and https://github.com/holke/t8code/actions/runs/2301750767 to see how it works.

### CODE REVIEW

File: scripts/pre-commit [hunk 1]
```diff
@@ -30,6 +30,7 @@
 # exit with non-zero status after issuing an appropriate message if
 # it wants to stop the commit.
 #
+# Copyright Johannes Holke 2020
```
Comment (holke) [inline scripts/pre-commit:33]:
```suggestion
```

**Comment by holke** [comment]
meanwhile we handled the indentation in a different PR.

This PR now only adds uploading of the configure and make check error logs as artifacts.

**Comment by holke** [comment]
I tested this with a failing configure: https://github.com/holke/t8code/actions/runs/2301835043

File: configure.ac [hunk 2]
```diff
@@ -30,6 +30,8 @@ echo "o---------------------------------------"
 echo "| Checking MPI and related programs"
 echo "o---------------------------------------"
 
+
+
```
Comment (holke) [inline configure.ac:34]:
```suggestion
```

File: config/t8_netcdf.m4 [hunk 3]
```diff
@@ -31,6 +31,7 @@ if test "x$T8_WITH_NETCDF" != xno ; then
 ]],[[
   #include <netcdf.h>
 
+  this line will break the code
```
Comment (holke) [inline config/t8_netcdf.m4:34]:
```suggestion

```

File: config/t8_netcdf.m4 [hunk 4]
```diff
@@ -31,6 +31,7 @@ if test "x$T8_WITH_NETCDF" != xno ; then
 ]],[[
   #include <netcdf.h>
```
Comment (holke) [inline config/t8_netcdf.m4:33]:
```suggestion

```

File: config/t8_netcdf.m4 [hunk 5]
```diff
@@ -30,7 +30,6 @@ if test "x$T8_WITH_NETCDF" != xno ; then
 [[
 ]],[[
   #include <netcdf.h>
```
Comment (holke) [inline config/t8_netcdf.m4:32]:
```suggestion
  #include <netcdf.h>
  
```

**Comment by holke** [comment]
> After one test failing, the github CI skips the remaining tests, but the OnFailUploadLog still gets triggered. They then fail themselves, because there in no log to upload, because the test never ran. It is not exactly a problem, but it does not look nice https://github.com/holke/t8code/runs/6373185043?check_suite_focus=true

Yes, i agree. We have this behavior from the beginning and its not directly related to uloading the logs.
Maybe there is a way to exit the action once it has failed...if so i could add it.

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -291,70 +291,125 @@ jobs:
...
-    - name: OnFailPrintLog
+    - name: OnFailUploadLog
       if: failure()
-      run: cat build_release/config.log
+      uses: actions/upload-artifact@v1
+      with:
+        name: config_release_MPI.log
+        path: build_release/config.log
     - name: make
       run: cd build_release && make -j V=0 CFLAGS="$CFLAGS_var -Werror" CXXFLAGS="$CXXFLAGS_var -Werror"
```
[the remark this answers is not in the document]
Comment (holke) [reply .github/workflows/tests.yml:301]: The `make` step does not produce any log files that we could upload.
We get the output of `make` in the console which we can look at in the browser.

[the remark this answers is not in the document]
Comment (sandro-elsweijer) [reply .github/workflows/tests.yml:301]: We could use `make > make_release_MPI.log 2>&1`. But this was just an idea, I do not know how relevant this is

[the remark this answers is not in the document]
Comment (holke) [reply .github/workflows/tests.yml:301]: I think its not necessary to do that.

**Comment by holke** [comment]
From my point of view this is done.
You can approve and merge :)

**Review by sandro-elsweijer — APPROVED** [review]

**Pull request merged**